spec/controllers/api/web/push_subscriptions_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Api::Web::PushSubscriptionsController do
+  render_views
+
+  let!(:user) { Fabricate(:user) }
+
+  let!(:create_payload) do
+    {
+      data: {
+        endpoint: 'https://fcm.googleapis.com/fcm/send/fiuH06a27qE:APA91bHnSiGcLwdaxdyqVXNDR9w1NlztsHb6lyt5WDKOC_Z_Q8BlFxQoR8tWFSXUIDdkyw0EdvxTu63iqamSaqVSevW5LfoFwojws8XYDXv_NRRLH6vo2CdgiN4jgHv5VLt2A8ah6lUX',
+        keys: {
+          p256dh: 'BEm_a0bdPDhf0SOsrnB2-ategf1hHoCnpXgQsFj5JCkcoMrMt2WHoPfEYOYPzOIs9mZE8ZUaD7VA5vouy0kEkr8=',
+          auth: 'eH_C8rq2raXqlcBVDa1gLg==',
+        },
+      }
+    }
+  end
+
+  let!(:alerts_payload) do
+    {
+      data: {
+        alerts: {
+          follow: true,
+          favourite: false,
+          reblog: true,
+          mention: false,
+        }
+      }
+    }
+  end
+
+  describe 'POST #create' do
+    before do
+      sign_in(user)
+
+      ActiveSession.deactivate warden.raw_session['auth_id']
```

Comment:
The tests fail because these two lines need to be executed after each call to `sign_in`. Tried putting the following into `spec_helper.rb` but didn't help. Any pointers?

```ruby
Devise.setup do |config|
  Warden::Manager.after_set_user except: :fetch do |user, auth, opts|
    ActiveSession.deactivate warden.raw_session['auth_id']
    warden.raw_session['auth_id'] = user.activate_session
  end
end
```